### PR TITLE
Fix video clips cutoff and json generation for them

### PR DESCRIPTION
**Pull request by yunikkk**

### CODE REVIEW

File: MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt [hunk 1]
```diff
@@ -199,37 +199,40 @@ internal interface SessionManager {
...
+                coreSessionStartMillis: Long
             ) {
                 if (currentCountry != Country.Unknown) {
 
-                    val sessionCountryDir = getCountryTelemetryDir() ?: return
-
-                    val sessionCountryDirWithTimeout = "$sessionCountryDir$sessionStartMillis/"
-
-                    moveFromCacheDirToCountryDir(outputPath, sessionCountryDirWithTimeout)
+                    val currentCountryTelemetryPath = getCountryTelemetryDir() ?: return
```
Comment (korshaknn) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]: maybe rename `getCountryTelemetryDir()` to `getCurrentCountryTelemetryPath()` as it always returns a path for current country

File: MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt [hunk 2]
```diff
@@ -240,12 +243,17 @@ internal interface SessionManager {
...
             }
         }
 
-        private fun moveFromCacheDirToCountryDir(fromDirPath: String, toDirPath: String) {
-            FileUtils.moveFiles(fromDirPath, toDirPath)
+        private fun generateSessionPath(
+            cachedTelemetryPath: String,
+            currentCountryTelemetryPath: String
+        ) : String {
+            val cachedPath = File((cachedTelemetryPath))
```
Comment (korshaknn) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]: useless braces?

Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]: Yup, will remove

File: MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt (hunk 1 above)
Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]: Okay

File: MapboxVision/src/main/java/com/mapbox/vision/utils/VisionLogger.kt [hunk 3]
```diff
@@ -156,7 +156,7 @@ interface VisionLogger {
 
     private class FileImpl(private val minLogLevel: Int = Log.INFO) : VisionLogger {
 
-        val fileWriter = FileWriter("${Environment.getExternalStorageDirectory()}/vision.log")
+        val fileWriter = FileWriter("${Environment.getExternalStorageDirectory()}/vision-${System.currentTimeMillis()}.log")
```
Comment (korshaknn) [inline MapboxVision/src/main/java/com/mapbox/vision/utils/VisionLogger.kt]: maybe write file with some DateTime format in name, not just long?

Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/utils/VisionLogger.kt]: Well, that makes sense - let's just move it out of this PR scope.

File: MapboxVision/src/main/java/com/mapbox/vision/video/videoprocessor/VideoProcessor.kt [hunk 4]
```diff
@@ -47,49 +47,47 @@ internal interface VideoProcessor {
...
+                    val relativeClipStartMillis = absoluteCoreClipStartMillis - coreSessionStartMillis
+                    val relativeClipEndMillis = absoluteCoreClipVideoEndMillis - coreSessionStartMillis
+                    if (relativeClipStartMillis < 0 || relativeClipEndMillis < 0) {
                         continue
                     }
-                    val timespan = "${"%.2f".format(Locale.ENGLISH, relativeStartMillis / 1000f)}_${"%.2f".format(
-                        Locale.ENGLISH,
-                        relativeEndMillis / 1000f
-                    )}"
+                    val timespan = "${part.endSeconds.formatSeconds()}_${part.startSeconds.formatSeconds()}"
```
Comment (korshaknn) [inline MapboxVision/src/main/java/com/mapbox/vision/video/videoprocessor/VideoProcessor.kt]: should be  `start_end` ?

Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/video/videoprocessor/VideoProcessor.kt]: Riiiight, good catch!

Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/video/videoprocessor/VideoProcessor.kt]: Fixed

**Pull request merged**